broadinstitute/single_cell_portal_core
# Add Metadata extraction from AnnData functionality (SCP-4824)

Changes: 
- Alters the way we pass flags for file extraction from AnnData files in line with the updates done in ingest pipeline. This affects cluster and metadata extraction
- Update the ingest pipeline to be the most recent version

Additions:
- Logic for metadata extraction from AnnData files

This adds the Rails side for the ingest pipeline update you can find here: https://github.com/broadinstitute/scp-ingest-pipeline/pull/287

To test:
Pre-reqs
- You will need a clean study with no files in it
- You will need to make sure your local ingest-pipeline configuration is not set or is set to the latest release of 1.24.0.
- You need to ensure you've got the feature flag for ingesting anndata set to true for your user and study
- download this h5ad file (and unzip it, just couldn't put here without compressing it): 
[compliant_pbmc3K.h5ad.zip](https://github.com/broadinstitute/single_cell_portal_core/files/10362369/compliant_pbmc3K.h5ad.zip)


Steps:
- Pull this branch
- Start up a local instance and also start up delayed jobs
- Go to your clean study that has no files in it
- Go to file upload and navigate to the AnnData upload
- Upload this file:
- Once the file completes go to the study details page
- You should see:
Cluster info:
![Screen Shot 2023-01-06 at 10 59 53 AM](https://user-images.githubusercontent.com/54322292/211054272-cd5c7407-01f7-42a1-9bcb-18bd4eaa7c65.png)
Metadata info:
![Screen Shot 2023-01-06 at 10 59 47 AM](https://user-images.githubusercontent.com/54322292/211054318-e6973922-cdc0-4d81-83bd-e32960aeb186.png)
And the AnnData file:
![Screen Shot 2023-01-06 at 11 54 17 AM](https://user-images.githubusercontent.com/54322292/211060020-eded3cb7-594b-4c32-9933-8f0f2e706cde.png)

You can also check out your studies google bucket to see the individual fragments:
![Screen Shot 2023-01-06 at 11 29 25 AM](https://user-images.githubusercontent.com/54322292/211054829-cfe90fcc-b702-4dbf-914e-b252f2ae857c.png)


https://user-images.githubusercontent.com/54322292/211059786-f5fbb124-3235-4062-aa3c-82f3d88c99eb.mov
